### PR TITLE
Add all-logs and rm options

### PR DESCRIPTION
**Pull request by pecigonzalo**
Related to: 
- #205 
- #206 
- #196

#### All Logs
As explained in #206, in many cases other container logs are required in order to properly troubleshoot a failed run.

#### `--rm` Option
Adding the `--rm` option and making it default to ensure you can run multiple runs of the same container or compose file.

#### Re-add linked container PS status
This allows for looking at the status before exit, I don't understand why it was removed in #196 as its quite useful for debugging a container being OOMd or similar.

## What
- Add `all-logs`
- Add `rm`
- Add linked container status output

### TODO
- [ ] Add tests
- [ ] Update docs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -417,6 +417,12 @@ Sets `docker-compose` to run with `--verbose`
 
 The default is `false`.
 
+### `all-logs` (optional)
+
+If set to false, only upload failed container logs.
+
+The default is `true`.
```
Comment (dan-kez) [inline README.md]: You probably want to add the rm option in the docs as well

Comment (pecigonzalo) [reply README.md]: Yep, it's on my TODO section. I wanted to check first if this was OK or not.

**Comment by pecigonzalo** [comment]
Please let me know if you are OK with this changes, so I proceed with the TODO

**Comment by toolmantim** [comment]
Sorry for the delay. I think the white space changes, and adding two options in the one pull request, was making it hard to figure out how to review this properly. Is there a chance you could do that? If not, it might have to wait until we do that for you and add any required changes.

**Comment by pecigonzalo** [comment]
Yeah, I can split the changes into 3 PRs.

**Comment by toolmantim** [comment]
Also I realized another thing that was making it tricky, not having any changes to the tests, which makes it hard to guess how it might affect existing users.

I think for both of these options we’ll want to keep our existing default bevhaioir. Is `rm` already true by default, but this gives you the option to disable it for your multiple runs?

**Comment by pecigonzalo** [comment]
I did not do the test as I did not know if this would be even a desired feature for you, so I only made the "simplest" changes to showcase the features.

The code does not do `rm` at the moment (except at cleanup) but as far as I can tell, the behavior doesn't change since we dont pull the logs for the main container, as they are printed in the console.

The other changes, as explained basically revert partially the behavior changes done in #196, which did a bunch of unlisted major changes.
eg. https://github.com/buildkite-plugins/docker-compose-buildkite-plugin/pull/196/commits/63a9cc1c7f8804cd8455f4aa875a4f943c451ab8 which is a complete change in how commands are passed, was just basically "wrapped" in a "cleanup output" PR, with no explanation whatsoever of the impact (basically you cant pass `do this && that` or multiple commands anymore because its taken as a single arg, IMHO, not a great choice).

**Comment by toolmantim** [comment]
I can understand the frustration of the changes not being well documented in the PR. Sorry about that! 

We’re happy to revisit changes, but they were made for good reasons and we need to think hard about tradeoffs — having clear before and after examples of how it broke things for you, and what subtleties these new PR changes might introduce on top of them, helps us all work together to design a way forward.

If you could split this into separate PRs with test changes (with no whitespace changes), that’d be much appreciated 🙏🏼 Then we could have better conversations and figure out the subtleties in each. And if not, that’s okay too, we’ll investigate getting all these issues looked at in good time.

**Comment by pecigonzalo** [comment]
Yeah, no worries at all, I'll split into 3 PR and close this one after. I'm
not trying to "bash" on the other PR but rather explaining my reasoning
here.
Given there seems to interest in the changes in happy to write the test and
other missing parts, I just don't start with that in case the changes are
not desired and to avoid the overhead while just showing the general idea.

**Comment by toolmantim** [comment]
I think both of these options make sense to add!

What about `upload-logs` with the valid values `”service”` (default), `”all”` and `false`? I think that matches current behavior, and gives you a new `upload-logs: “all”` option?

I’m trying to think through what `rm: true` would do, and what `rm: false` would do, and how it relates to current behavior. But let’s discuss in a new PR, so we don’t end up with the same confusing PR situation.

**Comment by toolmantim** [comment]
Or perhaps `upload-logs` with `”service”` (default), `true` (upload all logs), `false` (upload no logs)

**Comment by pecigonzalo** [comment]
Ill open 3 new PRs, and we can continue the discussion there.

**Comment by pecigonzalo** [comment]
@toolmantim all PRs open

**Comment by toolmantim** [comment]
Thanks @pecigonzalo! Sorry for the delay on the reviews 😓

**Pull request closed without merging**